modules/asset-manager-impl/src/main/java/org/opencastproject/assetmanager/impl/AssetManagerWithMessaging.java

```diff
@@ -229,9 +234,9 @@ public void notifyDeleteEpisode(String mpId) {
    * <p>
    * Do not call outside of a security context.
    */
-  private TakeSnapshot mkTakeSnapshotMessage(Snapshot snapshot) {
-    final AccessControlList acl = authSvc.getActiveAcl(snapshot.getMediaPackage()).getA();
-    return AssetManagerItem.add(workspace, snapshot.getMediaPackage(), acl, getVersionLong(snapshot),
+  private TakeSnapshot mkTakeSnapshotMessage(Snapshot snapshot, MediaPackage mp) {
+    final MediaPackage chosenMp = mp != null ? mp : snapshot.getMediaPackage();
```

Comment:
This type of conditional is frowned upon in the codebase, Entwine's initial code notwithstanding.

Reply:
Didn't know that, I'll rewrite it, then.